community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/CallProcedureAcceptanceTest.scala

```diff
@@ -0,0 +1,188 @@
+
+class CallProcedureAcceptanceTest extends ExecutionEngineFunSuite {
+
+  test("should be able to find labels from built-in-procedure") {
+    // Given
+    createLabeledNode("A")
+    createLabeledNode("B")
+    createLabeledNode("C")
+
+    //When
+    val result = execute("CALL sys.db.labels")
+
+    // Then
+    result.toList should equal(
+      List(
+        Map("label" -> "A"),
+        Map("label" -> "B"),
+        Map("label" -> "C")))
+  }
+
+  test("should be able to find types from built-in-procedure") {
+    // Given
+    relate(createNode(), createNode(), "A")
+    relate(createNode(), createNode(), "B")
+    relate(createNode(), createNode(), "C")
+
+    // When
+    val result = execute("CALL sys.db.relationshipTypes")
+
+    // Then
+    result.toList should equal(
+      List(
+        Map("relationshipTypes" -> "A"),
+        Map("relationshipTypes" -> "B"),
+        Map("relationshipTypes" -> "C")))
+  }
+
+  test("should be able to find propertyKeys from built-in-procedure") {
+    // Given
+    createNode("A" -> 1, "B" -> 2, "C" -> 3)
+
+    // When
+    val result = execute("CALL sys.db.propertyKeys")
+
+    // Then
+    result.toList should equal(
+      List(
+        Map("propertyKeys" -> "A"),
+        Map("propertyKeys" -> "B"),
+        Map("propertyKeys" -> "C")))
+  }
+
+  test("should be able to call procedure with explicit arguments") {
+    // Given
+    register(Neo4jTypes.NTString, Neo4jTypes.NTNumber)
+
+    // When
+    val result = execute("CALL my.first.proc('42', 42)")
+
+    // Then
+    result.toList should equal(List(Map("out0" -> "42", "out1" -> 42)))
+  }
+
+  test("should be able to call procedure with implicit arguments") {
+    // Given
+    register(Neo4jTypes.NTString, Neo4jTypes.NTNumber)
+
+    // When
+    val result = execute("CALL my.first.proc", "in0" -> "42", "in1" -> 42)
+
+    // Then
+    result.toList should equal(List(Map("out0" -> "42", "out1" -> 42)))
+  }
+
+  test("should fail if input type is wrong") {
+    // Given
+    register(Neo4jTypes.NTNumber)
+
+    // Then
+    a [CypherTypeException] shouldBe thrownBy(execute("CALL my.first.proc('ten')"))
+  }
+
+  test("if signature declares number all number types are valid") {
+    // Given
+    register(Neo4jTypes.NTNumber)
+
+    // Then
+    execute("CALL my.first.proc(42)").toList should equal(List(Map("out0" -> 42)))
+    execute("CALL my.first.proc(42.3)").toList should equal(List(Map("out0" -> 42.3)))
+  }
+
+  test("should fail a procedure declares an integer but gets a float ") {
+    // Given
+    register(Neo4jTypes.NTInteger)
+
+    // Then
+    a [CypherTypeException] shouldBe thrownBy(execute("CALL my.first.proc(42.0)"))
+  }
+
+  test("should fail a procedure declares a float but gets an integer") {
+    // Given
+    register(Neo4jTypes.NTFloat)
+
+    // Then
+    a [CypherTypeException] shouldBe thrownBy(execute("CALL my.first.proc(42)"))
+  }
+
+  test("should fail if explicit argument is missing") {
```

Comment:
Added test for having too many arguments. Tests for adding procedures are done at the kernel level, the only exposed way to create procedures will be from annotated java classes which are loaded when the database boots. The way it is done here is just a hack for tests.